config/v1/stable.infrastructure.testsuite.yaml

```diff
@@ -56,6 +56,120 @@ tests:
             external:
               platformName: SomeOtherCoolplatformName
       expectedError: " spec.platformSpec.external.platformName: Invalid value: \"string\": platform name cannot be changed once set"
+    - name: Should not be able to update External cloudControllerManager state once it was set
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec: {}
+        status:
+          platform: External
+          platformStatus:
+            type: External
+            external:
+              cloudControllerManager:
+                state: ""
+      updated: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec: {}
+        status:
+          platform: External
+          platformStatus:
+            type: External
+            external:
+              cloudControllerManager:
+                state: External
+      expectedStatusError: " status.platformStatus.external.cloudControllerManager.state: Invalid value: \"string\": state is immutable once set"
+    - name: Should not be able to add External cloudControllerManager state once it was set
```

Comment:
Do we have an issue with someone adding it to an existing cluster? I thought we allowed someone to add a CCM, but not remove it? 
By adding a CCM, they start benefiting from cloud provider specific details, such as annotations on new nodes for topology, or service of type load balancer.
But removing a CCM is undefined, how would it clean up services and their load balancers.

I think we went to prevent `External` -> `""` or removed, but, we should allow no CCM to start using one IMO

CC @elmiko do you agree?

Reply:
My understanding was to deny any transitions, transitioning from "None" to "External" would also mean that the existing nodes will never be initialized by the CCM unless we uninitialized them manually first (set the label `node.cloudprovider.kubernetes.io/uninitialized`).

It's a behavior we observed when we made tests with Oracle Cloud's CCM.

Reply:
this is a good question. my concern is that even with a None->External transition there would still need to be a roll out of the core kube components to adjust the `--cloud-provider` flag, and the user would need to coordinate the deployment of their CCM.

@JoelSpeed what is the official API stance on making this immutable now, but changing the mutability later if we get requests from users to enable the transition behavior, could we do that?

my preference would be to keep this immutable for now until we hear that people actually want to transition from None->External.

Reply:
Making it immutable now and enabling the transition later should be about the same amount of work as enabling the transition now. I can't think of a downside of deferring that part, other than we have to change the validation and documentation later.